### PR TITLE
Encapsulate scil_extract_dwi_shell.py

### PR DESCRIPTION
**Pull request by ppoulin91**
- Encapsulated `scil_extract_dwi_shell.py` in `utils/bvals_bvecs_tools.py`
  - I was unsure where to put this, maybe this could be moved elsewhere?
- Moved a volume block iterator to `image/utils.py`
- Tested on HCP data

### CODE REVIEW

**Comment by ppoulin91** [comment]
I rebased on master to resolve the confilcts

**Comment by AntoineTheb** [comment]
@arnaudbore @jchoude @GuillaumeTh I don't have write access but this is ready

**Comment by ppoulin91** [comment]
Rebased and fixed conflicts

File: scripts/scil_extract_dwi_shell.py [hunk 1]
```diff
@@ -32,7 +30,6 @@
 
 
 def build_args_parser():
-
     parser = argparse.ArgumentParser(
         formatter_class=argparse.RawTextHelpFormatter,
         description=DESCRIPTION)
```
Comment (GuillaumeTh) [inline scripts/scil_extract_dwi_shell.py]: Can you move the description at the beginning as in scil_resample_volume.py

**Pull request merged**